src/openforms/js/components/admin/form_design/registrations/zgw/ZGWOptionsForm.js

```diff
@@ -1,20 +1,117 @@
+  const confidentialityLevelChoices = getChoicesFromSchema(
+    zaakVertrouwelijkheidaanduiding.enum,
+    zaakVertrouwelijkheidaanduiding.enumNames
+  );
+
+  const numErrors = filterErrors(name, validationErrors).length;
+  const defaultGroup = apiGroupChoices.length === 1 ? apiGroupChoices[0][0] : undefined;
   return (
     <Field name={name} label={label}>
-      <ZGWFormFields
-        index={index}
-        name={name}
-        schema={schema}
-        formData={formData}
-        onChange={formData => onChange({formData})}
-      />
+      <>
+        <span style={{display: 'inline-flex', gap: '10px', alignItems: 'center'}}>
+          <button
+            type="button"
+            className="button"
+            onClick={e => {
+              e.preventDefault();
+              setModalOpen(true);
+            }}
+            // admin style overrides...
+            style={{
+              paddingInline: '15px',
+              paddingBlock: '10px',
+            }}
+          >
+            <FormattedMessage
+              description="Link label to open registration options modal"
+              defaultMessage="Configure options"
+            />
+          </button>
+
+          {numErrors > 0 && (
+            <ErrorIcon
+              text={intl.formatMessage(
+                {
+                  description: 'Objects API registration validation errors icon next to button',
+                  defaultMessage: `{numErrors, plural,
+              one {There is a validation error.}
+              other {There are {numErrors} validation errors.}
+            }`,
+                },
+                {numErrors}
+              )}
+              extraClassname="fa-lg icon icon--danger icon--compact icon--no-pointer"
+            />
+          )}
+        </span>
+
+        <FormModal
+          isOpen={modalOpen}
+          title={
+            <FormattedMessage
+              description="ZGW APIs registration options modal title"
+              defaultMessage="Plugin configuration: ZGW APIs"
+            />
+          }
+          closeModal={() => setModalOpen(false)}
+          extraModifiers={['large']}
+        >
+          <Formik
+            initialValues={{
+              ...formData,
+              // Ensure that if there's only one option, it is automatically selected.
+              zgwApiGroup: formData.zgwApiGroup ?? defaultGroup,
+            }}
+            onSubmit={(values, actions) => {
+              onChange({formData: values});
+              actions.setSubmitting(false);
+              setModalOpen(false);
+            }}
+          >
+            {({handleSubmit}) => (
+              <>
+                <ZGWFormFields
+                  index={index}
+                  name={name}
+                  apiGroupChoices={apiGroupChoices}
+                  confidentialityLevelChoices={confidentialityLevelChoices}
+                />
+
+                <SubmitRow>
+                  <SubmitAction
```

Comment:
Why do we need this since the `SubmitRow` contains `SubmitAction` and `event.preventDefault()`?It's a bit confusing to me..

Reply:
we need to make sure that Formik's `handleSubmit` gets called, since it will run (client-side) validation en then perform the `onSubmit` if everything is fine. So we don't let the browser default behaviour submit to the backend, as we need to handle everything in JS (that's what the `event.preventDefault()` is for).